File: leetcode.com/0034_find_first_and_last_position_of_element_in_sorted_array/solution.c
int* searchRange(int* nums, int numsSize, int target, int* returnSize)
{
    int min = 0, mid = 0, max = numsSize - 1;
    int* result = (int*) malloc(sizeof(int) * 2);
    *returnSize = 2;

    result[0] = -1;
    result[1] = -1;

    if (numsSize == 0)
        return result;

    while (true) {
        if (min > max)
            return result;

        mid = min + ((max - min) / 2);

        if (nums[mid] == target)
            break;
        else if (nums[mid] > target)
            max = mid - 1;
        else if (nums[mid] < target)
            min = mid + 1;
    }

    result[0] = mid;
    result[1] = mid;

    /* backward */
    mid = result[0];
    while (mid >= 0 && nums[mid] == target) {
        result[0] = mid--;
    }

    /* forward */
    mid = result[1];
    while (mid < numsSize && nums[mid] == target) {
        result[1] = mid++;
    }

    return result;
}

File: leetcode.com/0034_find_first_and_last_position_of_element_in_sorted_array/solution.go
func searchRange(nums []int, target int) []int {
    result := []int{-1,-1}

    for i, n := range nums {
        if n == target {
            result[0] = i
            result[1] = i

            for j := i + 1 ; j < len(nums) ; j++ {
                if nums[j] != target {
                    break
                }

                result[1] = j
            }

            break
        }
    }

    return result
}
